core/src/main/java/io/grpc/MethodDescriptor.java

```diff
@@ -147,20 +148,23 @@ public final boolean serverSendsOneMessage() {
   public static <RequestT, ResponseT> MethodDescriptor<RequestT, ResponseT> create(
       MethodType type, String fullMethodName,
       Marshaller<RequestT> requestMarshaller,
-      Marshaller<ResponseT> responseMarshaller) {
+      Marshaller<ResponseT> responseMarshaller,
+      int serviceIndex) {
```

Comment:
Are we concerned with breaking old generated code? If so, we'll want to make this a new overload. It may make sense in general to have the overload.